core-war/src/main/java/org/silverpeas/web/importexport/servlets/ImportDragAndDrop.java

```diff
@@ -91,11 +94,25 @@ public void doPost(HttpServletRequest req, HttpServletResponse res)
 
     UserDetail currentUser = UserDetail.getCurrentRequester();
     String userLanguage = currentUser.getUserPreferences().getLanguage();
+    String componentId = request.getParameter("ComponentId");
     UploadSession uploadSession = UploadSession.from(request);
+    var documentTemplate = ofNullable(request.getParameter("documentTemplateId"))
+        .filter(StringUtil::isDefined)
+        .map(DocumentTemplateWebManager.get()::getDocumentTemplate)
+        .map(t -> Pair.of(t, request.getParameter("fileName")))
+        .filter(p ->StringUtil.isDefined(p.getSecond()));
+    if (documentTemplate.isPresent()) {
+      uploadSession.forComponentInstanceId(componentId);
+      final DocumentTemplate docTpl = documentTemplate.get().getFirst();
+      final String fileName = documentTemplate.get().getSecond() + "." + docTpl.getExtension();
+      try (var i = docTpl.openInputStream()) {
+        uploadSession.getUploadSessionFile(fileName).write(i);
+      } catch (IOException e) {
+        throwHttpForbiddenError();
+      }
+    }
     StringBuilder result = new StringBuilder();
     try {
-      String componentId = request.getParameter("ComponentId");
-
       if (!uploadSession.isUserAuthorized(componentId)) {
         throwHttpForbiddenError();
       }
```

Comment:
There is one type object resulting of the ImportDragAndDrop : one or several publications according to context.
Each one linked to one or several attachments.
One attachment can be created from a content file directly (like previously) and now it can also be created from a document template.
So, there is not several methods here to implement in order to perform one way or an other.
Nevertheless, I refactored the class in order to remove all sonar feedback.

Reply:
ok, thank